JotunnLib/Managers/SynchronizationManager.cs

```diff
@@ -464,26 +465,102 @@ private void InvokeOnAdminStatusChanged()
             OnAdminStatusChanged?.SafeInvoke();
         }
 
+        /// <summary>
+        ///     Gets an IEnumerable of all default and custom config files that should be managed by SynchronizationManager. 
+        ///     Ignores config files for plugins with AdminOnlyStrictness == Low if Jotunn is not installed on the server.
+        /// </summary>
+        /// <returns></returns>
         private IEnumerable<ConfigFile> GetConfigFiles()
         {
             var loadedPlugins = BepInExUtils.GetDependentPlugins(true);
 
-            foreach (var plugin in loadedPlugins.Values)
+            foreach (BaseUnityPlugin plugin in loadedPlugins.Values)
+            {
+                if (ShouldManageConfig(plugin))
+                {
+                    yield return plugin.Config;
+                }
+            }
+            foreach (ConfigFile customConfigFile in CustomConfigs.Values)
+            {
+                if (ShouldManageConfig(customConfigFile))
+                {
+                    yield return customConfigFile;
+                }
+            }
```

Comment:
Have you checked that the timing is correct here? I.e. that when disconnecting from a server, plugins are unlocked?
Overall, I think the `ShouldManageConfig` check should rather be somewhere else, for example in the `LockConfigurationEntries` method (not limited to that). `GetConfigFiles` is used at places that are timing indipendent

Reply:
I did check the timings and seemed to work but I never checked how connecting to multiple servers with differing set ups within a single session worked and I could have missed something. I went for performing the check in `GetConfigFiles` since I thought it resulted in the least amount of duplicate code and effectively made it so Jotunn just didn't manage syncing a mod at all if Joutnn wasn't on the server.

Happy to change the check to be performing in the locking functions rather than in 'GetConfigFiles' though as it will probably prevent future issues and be an easier design to maintain and modify. 

I've since realized that I probably need to modify the `ShouldManageConfig` method to actually check if that specific mod is installed on the server though as simply having Jotunn installed on the server because some other mod needs it doesn't mean that users want to apply syncing to a mod that is only installed client-side.